libraries/chain/include/graphene/chain/market_object.hpp

```diff
@@ -125,12 +125,20 @@ class call_order_object : public abstract_object<call_order_object>
       share_type       debt;        ///< call_price.quote.asset_id, access via get_debt
       price            call_price;  ///< Collateral / Debt
 
+      optional<uint16_t> target_collateral_ratio; ///< maximum CR to maintain when selling collateral on margin call
+
       pair<asset_id_type,asset_id_type> get_market()const
       {
          auto tmp = std::make_pair( call_price.base.asset_id, call_price.quote.asset_id );
          if( tmp.first > tmp.second ) std::swap( tmp.first, tmp.second );
          return tmp;
       }
+
+      /// Calculate maximum quantity of collateral to sell and debt to cover to meet @ref target_collateral_ratio.
+      /// @return a pair of assets, the first item is collateral to sell, the second is debt to cover
+      pair<asset, asset> get_max_sell_receive_pair( price match_price,
```

Comment:
Actually, in the beginning, I only returned the first member; while coding I noticed that the second is needed, so changed it to return a pair; at last I found the first is not being used in main code, but is heavily used in test cases, in addition I'm not sure whether it will be useful for potential new APIs, so decided to leave it there.

Do you think it's necessary to change it to only return the second?

Reply:
Not strictly necessary, but a significant simplification IMO.

Reply:
Removed `pair`.